### DISCUSSION TITLE
Realtime losing some events

### DISCUSSION THREAD

**Discussion by panmona** [supabase/supabase#2527] [Questions]
While adding realtime to my application I noticed that sometimes Supabase doesn't receive some Realtimes events by inspecting it in the Network debugger.

I am aware that there are currently limitations to Realtime but they don't seem to apply to my case as there's not much usage of the database. This means: small amount of inserts per second (below 1), small amount of columns (realtime is enabled on one table with 8 columns).

Do you agree that this should not happen in such a low usage case? What information do you need me to provide for investigating this problem?

EDIT:
Also I noticed that the Websocket seems to stay open and still receive heartbeat events, even after unsubscribing from the channel and disconnecting from the client. Is this a bug?

**Comment by panmona** [comment]
@inian Could you maybe take a look at this?

**Comment by w3b6x9** [comment]
@panmau can you please email [support@supabase.io](support@supabase.io) with your project ref so that we can investigate?

> Also I noticed that the Websocket seems to stay open and still receive heartbeat events, even after unsubscribing from the channel and disconnecting from the client. Is this a bug?

It was a bug but it was fixed very recently. Please upgrade `@supabase/supabase-js` to the latest version and see if that fixes it.

**Reply by panmona** [reply]
I actually noticed this with the local version of Supabase but it happens with the deployed one aswell. So it doesn't seem specific to one project to me.

About the second thing: I'll do that.

**Comment by w3b6x9** [comment]
@panmau

> I actually noticed this with the local version of Supabase but it happens with the deployed one aswell. So it doesn't seem specific to one project to me.

Interesting...if possible, can you please provide a screen recording and attach it as part of the support email so we can better visualize what's going wrong for you both locally and deployed?
